qiskit_nature/second_q/mappers/bosonic_logarithmic_mapper.py

```diff
@@ -0,0 +1,238 @@
+
+    Below is an example of how one can use this mapper:
+
+    .. code-block:: python
+
+        from qiskit_nature.second_q.mappers import BosonicLogarithmicMapper
+        from qiskit_nature.second_q.operators import BosonicOp
+
+        mapper = BosonicLogarithmicMapper(max_occupation=2)
+        qubit_op = mapper.map(BosonicOp({'+_0 -_0': 1}, num_modes=1))
+
+    .. note::
+        Since this mapper truncates the maximum occupation of a bosonic state as represented in the
+        qubit register, the commutation relations after the mapping differ from the standard ones.
+        Please refer to Section 4, equation 22 of Reference [2] for more details.
+
+    References:
+        [1] Bo Peng et al., Quantum Simulation of Boson-Related Hamiltonians: Techniques, Effective
+        Hamiltonian Construction, and Error Analysis, Arxiv https://doi.org/10.48550/arXiv.2307.06580
+
+        [2] R. Somma et al., Quantum Simulations of Physics Problems, Arxiv
+        https://doi.org/10.48550/arXiv.quant-ph/0304063
+    """
+
+    def __init__(self, max_occupation: int) -> None:
+        # Compute the actual max occupation from the one selected by the user
+        self.number_of_qubits_per_mode: int = (
+            1 if max_occupation == 0 else math.ceil(np.log2(max_occupation + 1))
+        )
+        max_calculated_occupation = 2**self.number_of_qubits_per_mode - 1
+        if max_occupation != max_calculated_occupation:
+            logger.warning(
+                f"The user requested a max occupation of {max_occupation}, but the actual "
+                + f"max occupation is {max_calculated_occupation}."
+            )
+        super().__init__(max_calculated_occupation)
+
+    @property
+    def number_of_qubits_per_mode(self) -> int:
+        """The minimum number of qubits required to represent a bosonic mode given a max occupation."""
+        return self._number_of_qubits_per_mode
+
+    @number_of_qubits_per_mode.setter
+    def number_of_qubits_per_mode(self, num_qubits: int) -> None:
+        if num_qubits < 1:
+            raise ValueError(f"The number of qubits must be at least 1, and not {num_qubits}.")
+        self._number_of_qubits_per_mode: int = num_qubits
+
+    def _map_single(
+        self, second_q_op: BosonicOp, *, register_length: int | None = None
+    ) -> SparsePauliOp:
+        """Maps a :class:`~qiskit_nature.second_q.operators.SparseLabelOp` to a ``SparsePauliOp``.
+
+        Args:
+            second_q_op: the ``SparseLabelOp`` to be mapped.
+            register_length: when provided, this will be used to overwrite the ``register_length``
+                attribute of the operator being mapped. This is possible because the
+                ``register_length`` is considered a lower bound in a ``SparseLabelOp``.
+
+        Returns:
+            The qubit operator corresponding to the problem-Hamiltonian in the qubit space.
```

Comment:
Please add a section along the lines of the following:
```
Raises:
    ValueError: explanation...
```
The same should be done for the linear mapper.